templates/hpa.yaml

```diff
@@ -0,0 +1,35 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v2beta1
```

Comment:
HPAs seem to be a rapidly evolving feature, available in several different API groups (v1, v2beta1, v2beta2, v2). We have a similar issue in this chart with the Ingress resource -- to maintain support for a spread of Kubernetes versions, special consideration has to be given to which API version(s) are allowed in the cluster to which the chart is being installed.

In the case of Ingress, only two different API groups need to be supported (historically there are three different API groups for Ingress resources, but one of those has reached its EOL). For HPAs, at least three API versions seem to be concurrently active, so special consideration may be needed for each of `v2`, `v2beta2`, `v2beta1`, and possibly `v1`. That would, I imagine, result in a very messy file here. Maybe not, though.

This changeset seems fine, and in alignment with the general strategy of other aspects of this chart in the manner in which it is designed to be configured. My only concern, really, is compatibility. `v2beta1` is too specific and niche, but if this changeset were to be augmented for compatibility with all non-EOL K8s versions, that's all I'd need for a :+1:.

Reply:
Thank you for detailed feedback. 
I agree. I have just updated PR which allows to switch between version `autoscaling/v1` and `autoscaling/v2` with values. By default version is `autoscaling/v1` if `autoscale.enable` is set to `true`. I intentionally created two files to simplify maintenance and phase out when possible. If you prefer, I can also bundle things in same file because specifications as far as I can see are quite close.
If possible please give me a day or two to test this thoroughly.

Reply:
I managed to test chart on kubenetes `v1.21.8` (`autoscaling/v1`) and kubernetes `1.23.1`- minikube (`autoscaling/v2`)  and selection of _HPA_ works for me as intended.